File: backend/models/intelligence.py
from pydantic import BaseModel
from typing import List, Dict
from datetime import datetime

class Trend(BaseModel):
    title: str
    growth_percentage: float
    forecast: str
    keywords: List[str]
    sources: List[str]

class CompetitiveMove(BaseModel):
    company: str
    move: str
    date: datetime
    relevance: float

class AdvancedIntelligence(BaseModel):
    niche: str
    week: int
    trends: List[Trend]
    insights: str
    competitive_moves: List[CompetitiveMove]
    actionable_advice: str
    sentiment: str  # positive, cautious, negative
    data_summary: Dict[str, str]
    created_at: datetime

class AdvancedIntelligenceInDB(AdvancedIntelligence):
    id: str

File: frontend/src/types/intelligence.ts
export interface Trend {
  title: string;
  growth_percentage: number;
  forecast: string;
  keywords: string[];
  sources: string[];
}

export interface CompetitiveMove {
  company: string;
  move: string;
  date: string;
  relevance: number;
}

export interface AdvancedIntelligence {
  id?: string;
  niche: string;
  week: number;
  trends: Trend[];
  insights: string;
  competitive_moves: CompetitiveMove[];
  actionable_advice: string;
  sentiment: 'positive' | 'cautious' | 'negative';
  data_summary: Record<string, string>;
  created_at: string;
}
